File: 백준/Silver/11403. 경로 찾기/경로 찾기.js
const fs = require('fs');
const input = fs.readFileSync("/dev/stdin").toString().trim().split('\n');

let n = +input.shift();
let arr = [];

for(const row of input) {
    arr.push(row.split(" ").map(Number));
}

for(let k = 0; k < n; k++) {
    for(let i = 0; i < n; i++) {
        for(let j = 0; j < n; j++) {
            if(arr[i][k] && arr[k][j]) {
                arr[i][j] = 1;
            }
        }
    }
}

for(let i = 0; i<n; i++) {
    console.log(arr[i].join(" "));
}

File: 백준/Silver/11403. 경로 찾기/경로 찾기.cc
#include <iostream>

using namespace std;

int arr[101][101];

int main() {
	int n;
	cin >> n;
	
	for(int i=1; i<=n; i++) {
		for(int j=1; j<=n; j++) {
			cin >> arr[i][j];
		}
	}
	
	for(int k=1; k<=n; k++) {
		for(int i=1; i<=n; i++) {
			for(int j=1; j<=n; j++) {
				if(arr[k][j] && arr[i][k]) {
					arr[i][j] = 1;
				}
			}
		}
	}
	
	for(int i=1; i<=n; i++) {
		for(int j=1; j<=n; j++) {
			cout << arr[i][j] << " ";
		}
		cout << '\n';
	}
}